packages/medusa/src/payments/payment-service.ts

```diff
@@ -0,0 +1,93 @@
+import { Payment } from "@models/payment"
+import { Cart } from "@models/cart"
+import { PaymentSession, PaymentSessionStatus } from "@models/payment-session"
+import { Customer } from "@models/customer"
+import { BaseService } from "medusa-interfaces"
+
+export type Data = Record<string, unknown>
+export type PaymentData = Data
+export type PaymentSessionData = Data
+
```

Comment:
The three types above are the same but increase readability about there purpose. It will be easier for a user to understand the intelisens when seeing `PaymentSessionData` instead of just `Record`

Reply:
I think the types here are an excellent way of doing things! 🎉 

In regards to your other comment about naming of the methods: I think we need to stick with `create...` when we are initializing a payment session so that we are indicating to plugin authors that "this is the place you create something; i.e. it is okay to make a post request here".

If we call it `getPaymentSessionData` it feels like it is only okay to do retrieval of existing data. What are your thoughts on maybe naming it this way:

`createPayment` -> `createPaymentSessionData` or maybe even `onCreatePaymentSession` to indicate that it is more like a handle?

Reply:
> I think the types here are an excellent way of doing things! 🎉

I agree! Would it make sense to add `export type Data = Record<string, unknown>` to our shared, since I could imagine, that we would be using this typing pattern elsewhere.